More work on getting oracle demo and sdk tests working.

MicahZoltu opened the pull request:
* Adds require reason strings to all RLP require statements.
* Adds revert reason string to compiler generated reverts.
* Fixes a bug in the oracle where the tokens checked in the proof were backward.
* Adds a bunch of tests.

Commit:
Adds require statements to RLP reverts.

Commit:
Adds some uniswap swapping and time delays to demo.

Commit:
Contract bugfix (token order was wrong) and some more tests.

Commit:
Removes conditionals that cause the code to branch.

Branching messes up gas estimation, and these branches aren't necessary and don't save much gas, so don't do them.

epheph commented on contracts/source/UniswapOracle.sol:
```diff
@@ -66,16 +66,14 @@ contract UniswapOracle {
-					.uqdiv(denominationTokenIs0 ? reserve0 : reserve1)
-					) * secondsBetweenReserveUpdateAndHistoricBlock;
-			}
+			// TODO: figure out what denominationTokenIs0 means, re: reserve1/reserve0 ratios
```
As i mentioned in Discord, this one can be left in, because it doesn't change the estimate. the historic block will either always, or never, need to be "trued up"

epheph replied:
But leaving it out is fine too. saves 200 gas, costs maybe 15 to check?

MicahZoltu replied:
Added back.

epheph reviewed (approved):
I think we can add one of the conditionals back, but don't object to it staying out. The rest looks good, just gotta off-chain calculate the moving price

Commit:
Adds check back in.

The check is constant, not something that will mess with gas estimates.

Also re-ordered a couple of multiplications.
